### ISSUE TITLE
Running out of Monster IDs

### ISSUE THREAD

**Issue by Everade** [rathena/rathena#1253]
Monster IDs are usualy all set to **short** supporting up to 65,535.
(from what i've seen in the source code)

**However i'm not able to spawn any monsters using @monster command on a mob id range of 60k for example.**

Would be great if this could be increased as well.
As Gravity is keeping up quite fast by taking over all the ids.

Looking at the official lub files, they' already taking up the 6000range, also started up 10'000 and 25'000 range...
While rAthena at the current state only supports up to **3999** with the max mob id setting and the 4000-5000 being reserved by Player Clones.

I was recently forced to replace all custom item ids.
Now i have to do the same for monster ids.

And i can only tell it was probably the worst job i have ever done in my life.
Next to that, a few script functions were also not supporting Ids up to 65,535 (which i've reported recently)

So it would be great if you could increase the ids for all mob related things as well.
So we can make everything future proof.
Thank you.

So my guess is that at the current state, probably all private servers are running into the same issue as you can imagine.

**Comment by Everade** [comment]
bump

**Comment by aleos89** [comment]
If this is the case for various variable types used, then it will take some time to refactor the source. It won't be as easy as the item DB ID refactor.

**Comment by Everade** [comment]
It is the case.
And we've been slowly running into this issue.

In my point if view, it's getting kind of critial as we speak.
At least for most private servers.

**Comment by aleos89** [comment]
@secretdataz and I started chipping away at this the other day. It will take some time as it gets sort of boring and there are tons of variables that have to be checked because they aren't labeled properly.

**Comment by Everade** [comment]
With Episode 14 around the corner in the test branch updates.
They will start taking away the last available 3000 id range.

I assume that >50% of all servers who currently have custom mobs in use are within this range.

I'm not trying to rush you guys, i know it's a boring job. And there's probably more important things in your pipeline.
Just supposed to be another reminder.

**Comment by Jeybla** [comment]
> Monster IDs are usualy all set to short supporting up to 65,535.
> However i'm not able to spawn any monsters using @monster command on a mob id range of 60k for example.

Do you have any idea about the actual cause?

**Comment by Everade** [comment]
No idea.
But currently we still have the mob id 3999 limitation if i'm not mistaken.

I thought back then that i remind you early enough. But it seems that no one bothered updating it yet.

I stopped working on custom monsters since back then, so i'm still waiting for all the updates including all the script and ingame commands.

When we had the same issue with running out of item ids,  a lot of commands have been forgotten which ended up quite messy when you were running an actual server with new, higher item ids.

Maybe @aleos89 knows more since he started working on it over a year ago?

**Comment by Lemongrass3110** [comment]
The problem with item IDs was different. We used a signed data type, whereas kRO used a unsigned data type. That was why all commands had to be changed as well.

This is basically a problem of client side, because it has a hardcoded limit of < 4000. I can also tell you why and that is because 4001 is High Novice.
So since Gravity used up all mob ids now(see 6899c89 and https://github.com/rathena/rathena/blob/master/db/re/mob_db.txt#L3134) they will surely add the feature to their next sakray client(jRO already has it) as soon as they need to add monsters on sakray again. From then on we can rework the internal storage of monster ids and add support for the new 20.000+ range.

So to sum it up: This is not a problem we can overcome without the "help" of Gravity on client side.

**Comment by Lemongrass3110** [comment]
Small update:
Seems like kRO still had some IDs reserved for future usage, so they have not ran out of IDs on kRO yet.
See 705cc5a which merges the new IDs for the patch of 2017-10-11.

**Comment by Lemongrass3110** [comment]
Closed in b2ab972.
